appcelerator/titanium_mobile
# [TIMOB-26563] Android/iOS: Modified 2DMatrix translate() to use default units defined in "tiapp.xml"

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26563

**Summary:**
- This is a breaking-change.
- Android translate coordinates used to be hard-coded to pixels.
- iOS translate coordinates used to be hard-coded to dips.

**Note:**
Waiting for PR #10414 to be merged so that we can use its new iOS `convertDipToDefaultUnit()` function to resolve this commit's `TODO` comments. The `2DMatrix` "tx" and "ty" getters need to convert from dips back to default units so that they'll match what they were originally assigned to.

**Test:**
_Run the below test on both Android and iOS._
1. Create a classic app using the test code attached to [TIMOB-26563](https://jira.appcelerator.org/browse/TIMOB-26563).
2. Set "tiapp.xml" property "ti.ui.defaultunit" to `dp`.
3. Run the project.
4. On app startup, verify that a blue square moves from the top-left corner to the bottom-right corner of the window. _(Don't change app orientation. The test code won't compensate for it.)_
5. Set "tiapp.xml" property "ti.ui.defaultunit" to `px`.
6. Repeat steps 3-4.
